### PR TITLE
Fix accessing HTTPS sites with an IPv4 address

### PR DESCRIPTION
**Pull request by piotrklibert**
Hi, I was writing an internal monitoring tool and found that I need to access URLs in the form of `https://123.123.123.123/something/` - that is, URLs with HTTPS protocol and numeric IP in the host part. Unfortunately, when I tried issuing such requests I discovered that Hackney doesn't allow this case. (Actually, I was writing in Elixir with HTTPoison, which in turn uses Hackney.)

After some time debugging I discovered that Erlang's ssl application supports requests to IP addresses only if they are presented as four-tuples of ints, not as strings.

I created a quick fix for this which works for me - just before the `ssl:connect` I split the Host string on dots and check if they can be interpreted as an IP address. I'm not sure if it's any good, it's fine if it's not, but then I think it would be worth it to at least state in the README that HTTPS+IP doesn't work.

### CODE REVIEW

File: src/hackney_ssl.erl [hunk 1]
```diff
@@ -49,6 +49,13 @@ connect(Host, Port, Opts) ->
 
 connect(Host, Port, Opts, Timeout) when is_list(Host), is_integer(Port),
                                         (Timeout =:= infinity orelse is_integer(Timeout)) ->
+  Host1 = try
```
Comment (benoitc) [inline src/hackney_ssl.erl]: would be better to use  `inet:parse_address/1` there. Can you update your PR to use it? (sorry for the late answer)

Comment (piotrklibert) [reply src/hackney_ssl.erl]: Sure, no problem. Just to be sure, you mean something like this, right?
```
f(Host) ->
    Host1 = case inet:parse_address(Host) of
             {ok, Address} -> Address;
             {error, _} -> Host
         end,
    Host1.
```
Would it be better if I moved this to a separate function, or should it remain inline in `connect/4`?

I will probably update the PR when I get home in the evening (can't really code right now, family&stuff...)

Comment (benoitc) [reply src/hackney_ssl.erl]: maybe in a separate function to make it more readable.

Comment (benoitc) [reply src/hackney_ssl.erl]: @piotrklibert would be cool if you can send your commit today to be part of the release. Let me know :)

**Comment by piotrklibert** [comment]
Yes, it seems to be broken still for the IPv6 addresses (testing with Google's `2001:4860:4860::8888`). I don't immediately see why or where it happens, I will need some time to figure it out. IPv4 works (tested after the changes to PR).

**Comment by piotrklibert** [comment]
Scratch the above, it now kind of works, from what I understand as long as the `no_proxy_sni_support` here: https://github.com/benoitc/hackney/blob/6ef0f9bb2cdfa654643be8324e081bf7f05d659a/src/hackney_http_connect.erl#L27-L37 is defined (that is, when the `service_name_indication` is added to options).

**Comment by OvermindDL1** [comment]
Precisely the same issue here, I keep getting back an `%HTTPoison.Error{id: nil, reason: {:options, {{:server_name_indication, '10.0.1.2'}}`...

**Comment by OvermindDL1** [comment]
Tested other commits, it was broken on 6ef0f9bb2cdfa654643be8324e081bf7f05d659a.

**Comment by OvermindDL1** [comment]
And manually applying this patch did indeed fix it.  :-)

Any chance of a release?

**Comment by benoitc** [comment]
@OvermindDL1 thanks, sorry for the delay to apply it.

**Comment by OvermindDL1** [comment]
Awesome thanks!  Any chance on a hex.pm release so I don't have to depend on a git branch anymore?  ^.^

**Pull request merged**